packages/blocks/src/store/test/selectors.js

```diff
@@ -19,6 +17,16 @@ import {
 	getActiveBlockVariation,
 } from '../selectors';
 
+const keyBy = ( arr, key ) =>
+	( arr || [] ).reduce(
+		( r, x ) => ( { ...r, [ key ? x[ key ] : x ]: x } ),
+		{}
+	);
+const omit = ( obj, keys ) =>
+	Object.fromEntries(
+		Object.entries( obj ).filter( ( [ key ] ) => ! keys.includes( key ) )
+	);
```

Comment:
This is the kind of situation where I personally wouldn't be in favor of replacing lodash usage, because these functions are mostly simple to understand in terms of input and output, but the implementation is quite complex and they're only unit tested by proxy.

I do understand the overall goal here. I just wonder whether there's a better way. 

The first one is also not meeting the coding standards, where shortened names like `r` and `x` shouldn't be used in favour of a more descriptive alternative. That's at least easily fixable.

Reply:
Thanks for all the candid feedback! 

While this is in tests and has little impact, I think what you brought up is a valid concern to bring up and I'm happy to do another pass and simplify it. There's actually some opportunity for that here and I'm happy to grasp it.

Reply:
Just pushed a couple of new commits, and I'd love to hear what you think:

* I completely got rid of `omit()` in favor of a simpler declaration of the fixtures. It's unnecessary to declare a complex object and then pull it in pieces when we can just declare simpler objects in the first place.
* I refactored `keyBy` a bit to make it simpler in the current context, and more readable.

Let me know what you think, and thanks again!

Reply:
Oh, and I hope this new version helps you reconsider this statement:

> This is the kind of situation where I personally wouldn't be in favor of replacing lodash usage

😉

Reply:
Definitely better, thanks for addressing the feedback.

I still feel like some of these functions (e.g. isPlainObject) should be library code rather than having the implementation duplicated across the codebase.

Have there been any alternative explorations to this approach?

Reply:
Sounds good, thanks for replying to my concerns. Will give this one the thumbs up.